src/psyclone/tests/generator_test.py

```diff
@@ -80,13 +80,32 @@
                                 "test_files", "gocean1p0")
 
 
-def delete_module(modname):
-    '''A function to remove a module from Python's internal modules
-    list. This is useful as some tests affect others by importing
-    modules.
+@pytest.fixture(name="script_factory", scope="function")
+def create_script_factor(tmpdir):
+    ''' Fixture that creates a psyclone optimisation script given the string
+    representing the body of the script:
+
+        script_path = script_factory("def trans(psyir):\n  pass")
+
+    It has a 'function' scope and a tear down section because using a script
+    imports the file and this is kept in the python interpreter state, so we
+    delete it for future tests.
 
     '''
-    del modules[modname]
+    tmpfile = os.path.join(tmpdir, "test_script.py")
+
+    def populate_script(string):
+        with open(tmpfile, 'w+', encoding="utf8") as script:
+            script.write(string)
+        return tmpfile
```

Comment:
This returns `tmpfile` which is created outside of `populate_script()`. Is that OK?

Reply:
Yes, it is a [Closure](https://www.programiz.com/python-programming/closure). In the returned function I don't want it as a parameter but embedded inside fixed to the value provided by the fixture.

Reply:
Aha. I live and learn :-)